components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -488,20 +488,25 @@ void RewardsServiceImpl::GetContentSiteList(
   filter.non_verified = allow_non_verified;
   filter.min_visits = min_visits;
 
-  GetActivityInfoList(start, limit,
-      filter,
-      std::bind(&RewardsServiceImpl::OnGetContentSiteList,
-                this, callback, _1, _2));
+  bat_ledger_->GetActivityInfoList(
+      start,
+      limit,
+      filter.ToJson(),
```

Comment:
I think the `filter` creation might be something that belongs in `LedgerImpl`. I know it was already here, but if we're moving operations to the ledger side where they belong, we should probably move all the ledger ops there as well.

Reply:
well what if we want some different filter, we need to have a way to pass filter into ledger

Reply:
I think if you need to specify a different filter, that should be done in `Ledger`. In this case, it's created here and immediately passed in. Why would we need to pass a filter into Ledger? It should be Ledger's job to know what to tell `LedgerClient` what it wants.

Reply:
yes, filter defines what we want from ledger and we need to pass it right away . Creating function for every filter instance is not scalable for the same functionality

Reply:
Well, no, filter should define what you want from `LedgerClient` which I think should be dictated from Ledger. You don't need to create a function for every filter instance. Forward to a Ledger method called `LedgerImpl::GetContentSiteList` and then build your filter there with `CreateActivityFilter` (or an overload) and send it off to `LedgerClient`

Reply:
So let's say that UI needs to get this lists.
1) ac list for the current month
2) ac list for defined month
3) ac list for the current month sorted by % desc
4) ac list for the current month sorted by % asc

How would we do this without creating 4 rewards service functions?

Reply:
In `RewardsServiceImpl::GetContentSiteList`:
`bat_ledger_->GetActivityInfoList(start, limit, min_visit_time, reconcile_stamp, allow_non_verified, min_visits, base::oncecall);`

Then build filter in LedgerImpl

Reply:
not sure how this will not require 4 mojo calls for each scenario. Based on the fact that this was not introduced in this PR can we create issue for it and finalize flow there, so that this PR is not blocked on it?

Reply:
Sure. Maybe I'm not fully understanding what this is doing. I'll take to DM.